lib/aadutils.js

```diff
@@ -155,3 +156,21 @@ exports.rsaPublicKeyPem = (modulusB64, exponentB64) => {
 
   return pem;
 };
+
+// used for c_hash and at_hash validation
+// case (1): content = access_token, hashProvided = at_hash
+// case (2): content = code, hashProvided = c_hash
+exports.checkHashValueRS256 = (content, hashProvided) => {
+  // step 1. hash the content
+  var digest = crypto.createHash('sha256').update(content).digest();
```

Comment:
What if the value of 'content' is null? Will the update or digest method throw in that case? Also, createHash can throw if the OpenSSL version on platform does not support it: https://nodejs.org/api/crypto.html#crypto_crypto_createhash_algorithm, though that would be very unlikely  guess for sha256.